cmd/homeauth/homeauth.go

```diff
@@ -705,6 +715,41 @@ func (s *idpServer) loginUserSession(w http.ResponseWriter, r *http.Request, use
 		return err
 	}
 
+	// If the user wants to remember their username, set a cookie.
+	if r.FormValue("remember") == "on" {
+		s.logger.Debug("setting remember username cookie", "username", user.Email)
+		http.SetCookie(w, &http.Cookie{
+			Name:  rememberUsernameCookieName,
+			Value: user.Email,
+			Path:  "/",
+
+			// TODO(andrew-d): it feels a bit ugly to be reaching
+			// into the session manager for these fields; pull them
+			// out somewhere else?
+			Domain:   s.sessions.CookieOpts.Domain,
+			HttpOnly: s.sessions.CookieOpts.HttpOnly,
+			Secure:   s.sessions.CookieOpts.Secure,
+			SameSite: s.sessions.CookieOpts.SameSite,
+
+			// Set a reasonable upper limit for "remember me" of 30 days.
+			MaxAge: 30 * 24 * 60 * 60,
+		})
+	} else if cookie, err := r.Cookie(rememberUsernameCookieName); err == nil && cookie.Value != user.Email {
+		// If the user that's logging in is different from the one that's
+		// remembered, clear the cookie.
```

Comment:
Hm, if the user unchecks "remember me" but logs in with the remembered username, the cookie will remember. Seems unintuitive at first glance.

Reply:
Hm, good point. I updated this to instead clear the cookie (if it exists) on any login without "remember" checked.